blockchain/state_migration.go

```diff
@@ -371,3 +373,134 @@ func (bc *BlockChain) StopStateMigration() error {
 func (bc *BlockChain) StatusStateMigration() (bool, uint64, int, int, float64) {
 	return bc.db.InMigration(), bc.db.MigrationBlockNumber(), bc.committedCnt, bc.pendingCnt, bc.progress
 }
+
+func (bc *BlockChain) concurrentIterateTrie(root common.Hash, db state.Database, result chan common.Hash, finish chan error) (resultErr error) {
```

Comment:
Returned error is not handled.

Reply:
Also, it would be better to set the names of channels consistently. `result` and `finish` are named `resultHashCh` and `resultErrCh` in other places.

Reply:
> Returned error is not handled.

Below `resultErr` is used in below defer function.

Reply:
IMO, catching the returning variable complicates the function. How about removing the return variable like below?

```
func (bc *BlockChain) concurrentIterateTrie(root common.Hash, db state.Database, resultCh chan common.Hash, finishCh chan error) error {
  	stateDB, err := state.New(root, db)
 	if err != nil {
 		finishCh <- err
 	}

  	it := state.NewNodeIterator(stateDB)
 	for it.Next() {
 		resultCh <- it.Hash

  		select {
 		case <-bc.quitWarmUp:
 			finishCh <- errors.New("quitWarmUp")
 		case <-bc.quit:
 			finishCh <- errors.New("quit")
 		default:
 		}
 	}
 }
```

Reply:
After `finishCh<-err`, we need `return` always. 
So I wanted to simplify the code.
But as you talked, it is a bit complicated.
Even if we add several `return`, it's better than present code?